frontend/deb/distro/install.go

```diff
@@ -0,0 +1,174 @@
+package distro
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend/deb"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/pkg/errors"
+)
+
+// AptInstall returns an [llb.RunOption] that uses apt to install the provided
+// packages.
+//
+// This returns an [llb.RunOption] but it does create some things internally,
+// so any [llb.ConstraintsOpt] you want to pass in should come before this option
+// in a call to [llb.State.Run].
+func AptInstall(packages ...string) llb.RunOption {
+	return dalec.RunOptFunc(func(ei *llb.ExecInfo) {
+		const installScript = `#!/usr/bin/env sh
+set -ex
+
+# Make sure any cached data from local repos is purged since this should not
+# be shared between builds.
+rm -f /var/lib/apt/lists/_*
+apt autoclean -y
+
+apt update
+apt install -y $@
+`
+		script := llb.Scratch().File(
+			llb.Mkfile("install.sh", 0o755, []byte(installScript)),
+			dalec.WithConstraint(&ei.Constraints),
+		)
+
+		p := "/tmp/dalec/internal/deb/install.sh"
+		llb.AddMount(p, script, llb.SourcePath("install.sh")).SetRunOption(ei)
```

Comment:
What does `llb.SourcePath` do?

Reply:
Let selects a subpath, and in this caes lets us mount just the file.